### PR TITLE
[AV-991] Protobuf

### PR DESCRIPTION
**Pull request by zyunlam**
I haven't tested because I don't have the hardware for it, but it should work right out of the box.
https://trello.com/c/yIujqBtm

### CODE REVIEW

File: TARS/src/mcu_main/hilsim/generator/proto/__pycache__/__init__.cpython-311.pyc
Comment (gautamdayal) [inline TARS/src/mcu_main/hilsim/generator/proto/__pycache__/__init__.cpython-311.pyc:1]: Let's add this cache file extension to the gitignore

File: TARS/src/mcu_main/hilsim/hilsimpacket.proto [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * To generate hilsimpacket_pb2.py, use the following command:
+ * `protoc -I=. --python_out=. hilsimpacket.proto`
+*/
+syntax = "proto2";
+
+// TODO(Zyun): Might be more optimal to change all of this to required because protobuf for platformio
+// adds an extra bool between each data package, which could lead to some data wastage
+message HILSIMPacket {
+    // High-G IMU data
+    optional float imu_high_ax = 1;
```
Comment (gautamdayal) [inline TARS/src/mcu_main/hilsim/hilsimpacket.proto:19]: Can we get a comment to specify why the `optional` keyword is used here. It isn't immediately obvious to people that haven't worked extensively with pb (aka myself hehe)

File: TARS/src/mcu_main/sensors/LowGSensor.h [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #include "mcu_main/sensors/HighGSensor.h"
 #include "common/packet.h"
 #include "mcu_main/error.h"
-#include "mcu_main/hilsim/HILSIMPacket.h"
```
Comment (gautamdayal) [inline TARS/src/mcu_main/sensors/LowGSensor.h:8]: Seems like the only diff in the sensor abstraction layer is this. Looks like the update() overload didn't change? That's a good thing but can someone reply to this to confirm that and that I'm not just tripping? Thanks

File: TARS/src/mcu_main/hilsim/generator/proto/__pycache__/__init__.cpython-311.pyc
Comment (gautamdayal) [inline TARS/src/mcu_main/hilsim/generator/proto/__pycache__/__init__.cpython-311.pyc:1]: We might even be able to just add `generator/*` to the gitignore unless I'm mistaken

File: TARS/src/mcu_main/hilsim/hilsimpacket.pb.h
Comment (gautamdayal) [inline TARS/src/mcu_main/hilsim/hilsimpacket.pb.h:1]: Given that there's multiple files across which this serialization is defined, let's create a README within the HILSIM folder to outline the role of each one in the system

**Review by gautamdayal — COMMENTED** [review]
Good stuff! Mostly nitpicks from my side. Although definitely post an update on slack when this has been tested on flight hardware. Thanks for the work.

File: TARS/src/mcu_main/hilsim/generator/proto/__pycache__/__init__.cpython-311.pyc
Comment (zyunlam) [reply TARS/src/mcu_main/hilsim/generator/proto/__pycache__/__init__.cpython-311.pyc:1]: I added .pyc files to gitignore, and I swear I deleted them. They don't show up on my computer, which is weird

Comment (rbhog) [reply TARS/src/mcu_main/hilsim/generator/proto/__pycache__/__init__.cpython-311.pyc:1]: yeah u def can

Comment (gautamdayal) [reply TARS/src/mcu_main/hilsim/generator/proto/__pycache__/__init__.cpython-311.pyc:1]: Yeah it's weird sometimes just try it again. Maybe I accidentally pushed them back when I was trying to rebase

**Comment by aidancostello** [comment]
Merged with AV-991/999 PR, unnecessary

**Pull request closed without merging**